### ISSUE TITLE
Make 64K pages the default in Kconfig for Book3S/64

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#109]
Currently we have:

```
choice
	prompt "Page size"
	default PPC_4K_PAGES
```

For 64-bit BOOK3S the default should really be 64K, that's what modern CPUs are designed for.

It should be just a matter of:

```
	default PPC_64K_PAGES if PPC_BOOK3S_64
```

**And then we need to update the defconfigs to account for the flip.**

**Comment by shenki** [comment]
## Find the 64s machines
`PPC_BOOK3S_64` is the default for `PPC64`, so we need to find machines that are `PPC64` and set `PPC_BOOK3E_64`.

```
$ git grep -l PPC64=y arch/powerpc/configs/ 
arch/powerpc/configs/64-bit.config
arch/powerpc/configs/85xx-64bit.config
arch/powerpc/configs/cell_defconfig
arch/powerpc/configs/g5_defconfig
arch/powerpc/configs/maple_defconfig
arch/powerpc/configs/microwatt_defconfig
arch/powerpc/configs/pasemi_defconfig
arch/powerpc/configs/powernv_defconfig
arch/powerpc/configs/ppc64_defconfig
arch/powerpc/configs/ppc64e_defconfig
arch/powerpc/configs/ppc64le.config
arch/powerpc/configs/ps3_defconfig
arch/powerpc/configs/pseries_defconfig
arch/powerpc/configs/skiroot_defconfig
joel@voyager ~/dev/kernels/linux-next :(801a837d7533...) 
$ git grep -l PPC64=y arch/powerpc/configs/ | xargs git grep -l PPC_BOOK3E_64=y
arch/powerpc/configs/85xx-64bit.config
arch/powerpc/configs/ppc64e_defconfig
```

## Configs we are interested in
```
arch/powerpc/configs/64-bit.config
arch/powerpc/configs/cell_defconfig
arch/powerpc/configs/g5_defconfig
arch/powerpc/configs/maple_defconfig
arch/powerpc/configs/microwatt_defconfig
arch/powerpc/configs/pasemi_defconfig
arch/powerpc/configs/powernv_defconfig
arch/powerpc/configs/ppc64_defconfig
arch/powerpc/configs/ppc64le.config
arch/powerpc/configs/ps3_defconfig
arch/powerpc/configs/pseries_defconfig
arch/powerpc/configs/skiroot_defconfig
```
64-bit.config and ppc64le used to seed dynamically generated configs so they can be ignored.

```
$ cat configs.txt
cell
g5
maple
microwatt
pasemi
powernv
ppc64
ps3
pseries
skiroot
```
## Generate .configs from existing defconfigs
```
for config in `cat configs.txt`; do ARCH=powerpc make O=$config ${config}_defconfig; done
```
## Apply the change
```
git am 0001-powerpc-Make-64K-pages-the-default-for-Book3S.patch
```
## Generate new configs
```
for config in `cat configs.txt`; do ARCH=powerpc make O=${config}_64k ${config}_defconfig; done
```
## Show differences
```
for config in `cat configs.txt`; do diff -upqs ${config}/.config ${config}_64k/.config; done 

Files cell/.config and cell_64k/.config are identical
Files g5/.config and g5_64k/.config differ
Files maple/.config and maple_64k/.config differ
Files microwatt/.config and microwatt_64k/.config differ
Files pasemi/.config and pasemi_64k/.config are identical
Files powernv/.config and powernv_64k/.config are identical
Files ppc64/.config and ppc64_64k/.config are identical
Files ps3/.config and ps3_64k/.config differ
Files pseries/.config and pseries_64k/.config are identical
Files skiroot/.config and skiroot_64k/.config are identical
```

## Add 4K to the configs that are identical
This can be done by doing a savedefconfig on the old configuration, which has `CONFIG_PPC_4K_PAGES=y`.

In hindsight that would have been an easier method: create the .configs using the defconfigs, modify the default in Kconfig, and savedefconfig all of the old config files.

The downside of this approach is the noise from regenerating the defconfigs, so perhaps do that first.

**Comment by shenki** [comment]
https://lore.kernel.org/all/20211013233133.728484-1-joel@jms.id.au/

**Comment by shenki** [comment]
https://lore.kernel.org/r/20211015001649.45591-1-joel@jms.id.au

Merged in v5.16-rc1 f22969a6604165df658b99221b0aeb6918a9026d
